src/webapi/html_elements/canvas.rs

```diff
@@ -0,0 +1,107 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webcore::once::Once;
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::node::{INode, Node};
+use webapi::element::{IElement, Element};
+use webapi::html_element::{IHtmlElement, HtmlElement};
+use webapi::blob::Blob;
+
+/// The HTML `<canvas>` element provides an empty graphic zone on which specific JavaScript APIs
+/// can draw (such as Canvas 2D or WebGL).
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement)
+pub struct CanvasElement( Reference );
+
+impl IEventTarget for CanvasElement {}
+impl INode for CanvasElement {}
+impl IElement for CanvasElement {}
+impl IHtmlElement for CanvasElement {}
+
+reference_boilerplate! {
+    CanvasElement,
+    instanceof HTMLCanvasElement
+    convertible to EventTarget
+    convertible to Node
+    convertible to Element
+    convertible to HtmlElement
+}
+
+/// Trait implemented by rendering contexts which can be obtained from a canvas.
+pub trait RenderingContext {
+    /// Name which identifies this kind of rendering context.
+    fn from_canvas(canvas: &CanvasElement) -> Option<Self> where Self: Sized;
```

Comment:
I think it would make more sense for this to return `Result<Self, Self::Error>`.

Reply:
Yeah